src/components/QRShare/index.js

```diff
@@ -0,0 +1,98 @@
+import React, {Component} from 'react';
+import QRCodeLibrary from 'react-native-qrcode-svg';
+import {View} from 'react-native';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import defaultTheme from '../../styles/themes/default';
+import styles from '../../styles/styles';
+import Text from '../Text';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+import compose from '../../libs/compose';
+import variables from '../../styles/variables';
+import ExpensifyWordmark from '../../../assets/images/expensify-wordmark.svg';
+import {qrSharePropTypes, qrShareDefaultProps} from './propTypes'
+
+const propTypes = {
+    ...qrSharePropTypes,
+    ...windowDimensionsPropTypes,
+    ...withLocalizePropTypes,
+};
+
+class QRShare extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            qrCodeSize: 0,
+        };
+
+        this.onLayout = this.onLayout.bind(this);
+        this.getSvg = this.getSvg.bind(this);
+    }
+
+    onLayout(event) {
+        this.setState({
+            qrCodeSize: event.nativeEvent.layout.width - variables.qrShareHorizontalPadding * 2,
+        });
+    }
+
+    getSvg() {
+        return this.svg;
+    }
+
+    render() {
+        return (
+            <View
+                style={styles.shareCodeContainer}
+                onLayout={this.onLayout}
+            >
+                <View
+                    style={{
+                        alignSelf: 'stretch',
+                        height: 27,
+                        marginBottom: 20,
+                    }}
+                >
+                    <ExpensifyWordmark
+                        fill={defaultTheme.borderFocus}
+                        width="100%"
+                        height="100%"
+                    />
+                </View>
+
+                <QRCodeLibrary
+                    value={this.props.url}
+                    logo={this.props.logo}
+                    getRef={(svg) => (this.svg = svg)}
+                    logoBackgroundColor="transparent"
+                    logoSize={this.state.qrCodeSize * 0.3}
+                    logoBorderRadius={this.state.qrCodeSize}
+                    size={this.state.qrCodeSize}
+                    backgroundColor={defaultTheme.highlightBG}
+                    color={defaultTheme.text}
+                />
+
+                <Text
+                    family="EXP_NEW_KANSAS_MEDIUM"
+                    fontSize={22}
+                    style={{marginTop: 15}}
+                >
+                    {this.props.title}
+                </Text>
+
+                {this.props.subtitle && (
```

Comment:
Because we didn't use boolean for conditional render, it caused a regression (although dev), [here](https://github.com/Expensify/App/issues/19985). It is essential for conditional render to use boolean expressions to avoid console errors.

>  I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. myBool && <MyComponent />.

We also have an item in the checklist which was marked but we missed highlighting this one.

Reply:
Got it, thanks for the feedback! 🙇